### PR TITLE
Bind expanded extents in executor

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This is to support inputs with expanded extent, see `FusionExpandedInput_CUDA`

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/evaluator_common.cpp [hunk 1]
```diff
@@ -97,6 +97,11 @@ std::vector<Val*> collectRuntimeUsedValues(kir::Kernel* kernel) {
     for (auto id : tv->domain()->domain()) {
       ret.push_back(id->extent());
     }
+    for (auto id : tv->getMaybeRFactorDomain()) {
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/evaluator_common.cpp:100]: Ha, this looks promising, let me try this fix on my other problem: https://github.com/csarofeen/pytorch/issues/2053

File: torch/csrc/jit/codegen/cuda/evaluator_common.cpp [hunk 2]
```diff
@@ -418,9 +423,16 @@ void KernelPrecomputedValues::bindTensorMetaData(
       "Something went wrong configuring launch. Inputs do not match.");
 
   for (const auto dim : c10::irange(root_domain.size())) {
-    auto extent = root_domain[dim]->extent();
     auto value = tensor_arg_abstract->getSize(dim);
-    bindValue(extent->evaluatorIndex(), value);
+    if (root_domain[dim]->hasExpandedExtent()) {
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/evaluator_common.cpp:427]: so this part should only save bandwidth? Does it also properly propagate the expanded extent to output tensor as well?

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp [hunk 3]
```diff
@@ -842,6 +842,10 @@ void bindInputForExprEvaluation(
                 " but recieved value of ",
                 tensor_arg_size);
           }
+          auto expanded_extent = root_domain[dim]->expandedExtent();
+          if (!expanded_extent->isConstScalar()) {
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/executor_utils.cpp]: nitpick, if expanded_extent is const scalar, why do we need to bind the value again here?!
I'm somewhat expecting us to check that the expanded extent to match like the TORCH_CHECK above.

**Review by jjsjann123 — COMMENTED** [review]
A trivial question.

File: torch/csrc/jit/codegen/cuda/evaluator_common.cpp (hunk 2 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/evaluator_common.cpp:427]: It should. We are actually already able to "propagate" the expanded extent to outputs without this PR, if the expanded extent is bound somewhere (for example, as a fusion input, or as a shape of some other tensor in fusion input). This PR just makes sure that the expanded extents of a tensor is correctly bound when they should be.

File: torch/csrc/jit/codegen/cuda/evaluator_common.cpp (hunk 1 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/evaluator_common.cpp:100]: Nope... you didn't save me there :cry:

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp (hunk 3 above)
Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/executor_utils.cpp]: Good catch, I actually just need to replace this line and above with ` else {`

**Review by jjsjann123 — APPROVED** [review]
LGTM.

**Pull request merged**